.dagger/main.go

```diff
@@ -125,7 +125,19 @@ func (dev *DaggerDev) containing(ctx context.Context, filename string) ([]string
 
 // Dagger's Go toolchain
 func (dev *DaggerDev) Go() *GoToolchain {
-	return &GoToolchain{Go: dag.Go(dev.Source())}
+	// Modify the standard Go container by installing the protoc binary
```

Comment:
Hm 🤔 as mentioned in the other thread, I'm a touch worried about how this will work for the case of building an Ubuntu container (which is how our gpu images are built). Packages aren't necessarily just overlayable like this always.

That said - I'm not sure the best way to do this. I think having overlays isn't necessarily *wrong* but overlaying wolfi packages into operating systems that aren't wolfi doesn't feel right to me. And potentially could give scary errors later that are *very* hard to debug.

Could we overlay a single file, statically compiled protoc package? That would work in more cases. And also keeps the overlays, which I think is an elegant solution that splits out protoc from the concerns of the go package.

My main concern - I want to avoid the "use at your own risk" comment - it makes sense, and it's a good warning to have - but for our golden example CI that folks might copy-paste code and ideas from, I think we should try and avoid foot guns as much as possible.

Open to being persuaded! But as is, I do think this deserves some more discussion and bikeshedding 🚲🚲 (since it sets a new design and pattern, that we might use in other places, I think it's worth thinking it through together).